charms/benchmark/__init__.py

```diff
@@ -118,9 +120,26 @@ def start():
         If the cabs-collector charm is installed, take a snapshot
         of the current profile data.
         """
+        # Do profile data collection immediately on this unit
         if os.path.exists(COLLECT_PROFILE_DATA):
             subprocess.check_output([COLLECT_PROFILE_DATA])
 
+        # Tell the benchmark-gui charm the action_id via the benchmark
+        # relation. Benchmark-gui will pass the action_id to all
+        # collector charms in the environment (via the collector relation),
+        # triggering profile data collection on each.
+        if os.environ['CHARM_DIR']:
+            f = open(os.environ['CHARM_DIR'] + '/metadata.yaml', 'r')
```

Comment:
This block could be replaced using:

```python
with open(os.path.join(os.environ['CHARM_DIR'], 'metadata.yaml'), 'r') as f:
    metadata = yaml.safe_load(f.read())
```

which will take care of the closing of the file pointer for you